azure-pipelines.yml

```diff
@@ -248,13 +233,11 @@ jobs:
       )
     matrix:
       debian_atlas_32bit:
-        DISTRIB: 'debian-32'
         DOCKER_CONTAINER: 'i386/debian:11.2'
-        JOBLIB_VERSION: 'min'
+        DISTRIB: 'debian-32'
+        LOCK_FILE: './build_tools/azure/debian_atlas_32bit_lock.txt'
         # disable pytest xdist due to unknown bug with 32-bit container
         PYTEST_XDIST_VERSION: 'none'
```

Comment:
`PYTEST_XDIST_VERSION` is still used in the test script:
https://github.com/scikit-learn/scikit-learn/blob/7bdd426704fc49316b3273fb33de7f6931c22692/build_tools/azure/test_script.sh#L65-L66

Longer term, I was thinking about cleaning up environment variables like `PYTEST_XDIST_VERSION` and `COVERAGE` and check whether a package is installed instead of relying on a environment variable.